### PR TITLE
feat: add French translation to Kitsu

### PR DESCRIPTION
**Pull request by Arno500**
Changes proposed in this pull request:

- Add French to language selector on settings page,
- Add French localization file

I'll email Josh in a second
/cc @hummingbird-me/staff

### CODE REVIEW

File: app/controllers/settings/index.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ export default Controller.extend({
   notify: service(),
   lastUsed: storageFor('last-used'),
   user: alias('session.account'),
-  languages: getter(() => [{ id: 'en', text: 'English' }]),
+  languages: getter(() => [{ id: 'en', text: 'English' }, { id: 'fr', text: 'French' }]),
```
Comment (wopian) [inline app/controllers/settings/index.js]: Should be Français 👍

Comment (Arno500) [reply app/controllers/settings/index.js]: Yeah, mentionned it in the commit, would probably be better to localize languages, on the current one. But changing to "Français" for now.

File: app/instance-initializers/ember-intl.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 export function initialize(appInstance) {
   const intl = appInstance.lookup('service:intl');
-  intl.setLocale('en-us');
+  intl.setLocale('fr-fr');
```
Comment (trmcnvn) [inline app/instance-initializers/ember-intl.js]: Could you revert this change?

**Review by trmcnvn — DISMISSED** [review]
This is awesome! Great work @Arno500 👍 

We'll have to add some code that allows loading each translation asynchronously rather than bundling both and load the correct one based on the users choice 😄

Did you want to attempt that? If not, I'm happy to jump in some time this weekend and finish it off.

File: app/instance-initializers/ember-intl.js (hunk 2 above)
Comment (Arno500) [reply app/instance-initializers/ember-intl.js]: Oopsie, commited a bit too fast ! Reverting !

**Comment by Arno500** [comment]
> This is awesome! Great work @Arno500 👍
> 
> We'll have to add some code that allows loading each translation asynchronously rather than bundling both and load the correct one based on the users choice 😄
> 
> Did you want to attempt that? If not, I'm happy to jump in some time this weekend and finish it off.

Thanks a lot !
This is probably the best solution.
I'm more a React guy for now, and I don't know at all how Kitsu is working and initialized, so I'll leave it to you 👍, for the best possible implementation.

Also as talked with Josh, some keys are missing. You can expect some tasks of "refactoring" soon I think xD.

**Comment by Arno500** [comment]
@trmcnvn Did you have some time to spare on locale loading ? Only a friendly up :P

**Comment by wopian** [comment]
Rebased

File: translations/fr-fr.yaml [hunk 3]
```diff
@@ -0,0 +1,1280 @@
...
+        full_story: "Full story"
+        spinoff: "Spin-off"
+        adaptation: "Adaptation"
+        anime_adaptation: "Animé"
+        manga_adaptation: "Manga"
+        character: "Personnage"
+        other: "Autre"
+  hide-posts:
+    text: "Masquer les posts concernant les {media}"
+    info: "Les posts n'apparaîtrons plus dans vos fils"
```
Comment (wopian) [inline translations/fr-fr.yaml]: ICU argument missing: `{type}`


`en-us.yaml`:
```yml
info: "Posts won't appear in your Following or {type} Feed"
```

File: translations/fr-fr.yaml [hunk 4]
```diff
@@ -0,0 +1,1280 @@
...
+  show-posts:
+    text: "Voir les posts concernant les {media}"
+    info: "Les posts apparaîtrons dans vos fils"
```
Comment (wopian) [inline translations/fr-fr.yaml]: ICU argument missing: `{type}`


`en-us.yaml`:
```yml
info: "Posts will appear in your Following or {type} Feed"
```

File: translations/fr-fr.yaml [hunk 5]
```diff
@@ -0,0 +1,1280 @@
...
+        header: "Personnages"
+        more: "Voir tous les personnages"
+      titles:
+        header: "Titres Alternatifs"
+        en: "Anglais"
+        en_us: "Anglais (US)"
+        en_jp: "Romanisé"
+        ja_jp: "Japonais"
+        en_cn: "Pinyin"
+        zh_cn: "Chinois"
```
Comment (wopian) [inline translations/fr-fr.yaml]: Chinese has been changed to Simplified Chinese.

`zh_tw` key has been added for Traditional Chinese

**Review by wopian — CHANGES_REQUESTED** [review]

**Comment by Arno500** [comment]
Fixed, but the only problem is that I don't know what values can {type} have. This makes my translation imprecise for this part.
That's why I translated to something like "Posts won't appear in your feeds anymore" (French specifying plural in both the pronoun and the noun). There was still a loss of informations, though

**Comment by wopian** [comment]
@Arno500 `{type}` is Anime or Manga ~~(but in French)~~

Edit: I was wrong, not localised.

**Comment by wopian** [comment]
Adding the following in another PR and added here but not translated:
- German
- Spanish
- Italian
- Dutch
- Portuguese
- Portuguese (Brazilian) 
- Ukranian

**Comment by Arno500** [comment]
The problem is now the following:
- When `{type}` is Anime (animé in French), it should be 

> fil d'animé

- When `{type}` is Manga (manga in French), it should be

> fil de manga

The moving part "de" or "d'" is not the same according to the following word. How can I do that properly without destroying the other translations ?

**Comment by wopian** [comment]
@Arno500 the following works:

```yaml
info: "Les posts n'apparaîtrons plus ni dans votre fil d'abonnements{type, select,
          Anime {, ni fil d'animé}
          Manga {, ni fil de manga}
          other {}}"
```

EDIT: Updated code example: animés -> Anime and mangas -> Manga and moved `ni` into the switch so unknown types (e.g whenever Drama is added) doesn't have a dangling "or"

**Comment by wopian** [comment]
image

image

**Review by wopian — APPROVED** [review]
Good to merge if my fix in the last commit was OK

**Comment by Arno500** [comment]
Well I just small-fixed it, now we're good to go!
(there is still quite a lot of key missing actually, but I can't convert every module)

File: translations/fr-fr.yaml [hunk 6]
```diff
@@ -371,15 +371,15 @@ media:
         other: "Autre"
   hide-posts:
     text: "Masquer les posts concernant {media}"
-    info: "Les posts n'apparaîtrons plus ni dans votre fil d'abonnements{type, select,
-          Anime {, ni fil d'animé}
-          Manga {, ni fil de manga}
+    info: "Les posts n'apparaîtrons plus ni dans votre fil d'abonnements, ni {type, select,
```
Comment (wopian) [inline translations/fr-fr.yaml]: This produces the following (note the 3rd, ignore the `fil` as it was an unintentional addition):

- Les posts n'apparaîtrons plus ni dans votre fil d'abonnements, ni d'animés
- Les posts n'apparaîtrons plus ni dans votre fil d'abonnements, ni de mangas
- Les posts n'apparaîtrons plus ni dans votre fil d'abonnements, ni 

Previous produced:

- Les posts n'apparaîtrons plus ni dans votre fil d'abonnements, ni fil d'animé
- Les posts n'apparaîtrons plus ni dans votre fil d'abonnements, ni fil de manga
- Les posts n'apparaîtrons plus ni dans votre fil d'abonnements

Comment (Arno500) [reply translations/fr-fr.yaml]: Yes, forgot about the other, correcting

**Review by wopian — APPROVED** [review]

**Comment by wopian** [comment]
>(there is still quite a lot of key missing actually, but I can't convert every module)

Yeah...the translation file is way bigger than I expected it to be when I opened it yesterday.

**Comment by Arno500** [comment]
It took me 2 days to translate the entire file and check the results on the interface x)
I think approx. 15-20% of the content is not in the translation file currently

**Review by trmcnvn — APPROVED** [review]

**Comment by Arno500** [comment]
You may also consider using a tool like Transifex or Lokalise to allow easier editing and management. (to also prevent a lot of commits)
And it's easier for the community to contribute ;)

**Comment by wopian** [comment]
Will likely be using Pontoon (Mozilla) in the future 🤞

**Comment by wopian** [comment]
French is live 🎉

**Pull request merged**